tests/testthat/test-detect_column.R

```diff
@@ -0,0 +1,122 @@
+context("detect_column")
```

Comment:
Overall, this looks good, but I'm going to wait for a more detailed review for when the above comments are addressed.